rules/opentitan.bzl

```diff
@@ -726,12 +771,107 @@ pick_correct_archive_for_device = rv_rule(
     toolchains = ["@rules_cc//cc:toolchain_type"],
 )
 
+def opentitan_multislot_flash_binary(
+        name,
+        srcs,
+        image_size,
+        key_per_binary,
+        offset_per_binary,
+        platform = OPENTITAN_PLATFORM):
+    """A helper macro for generating multislot OpenTitan binary flash images.
+
+    This macro is mostly a wrapper around the `assemble_flash_image` rule, that
+    invokes `opentitantool` to stitch together multiple `opentitan_flash_binary`
+    images to create a single image for bootstrapping. This enables efficient
+    testing by only requiring one boostrap operation to load both silicon
+    creator and owner stages of flash.
+    Args:
+      @param name: The name of this rule.
+      @param srcs: A list of `opentitan_flash_binary` targets to stich together.
+      @param image_size: The final flash image_size to pass to `opentitantool`.
+      @param key_per_binary: A dictionary mapping `opentitan_flash_binary`
+                             targets to the signing key names that were used for
+                             signing each target.
+      @param offset_per_binary: A dictionary mapping `opentitan_flash_binary`
```

Comment:
nit: Is there a particular reason why you chose to separate `srcs`, `key_per_binary` and `offset_per_binary`. I feel like passing a dict per flash_binary with all these grouped together would be easier to follow.

Reply:
updated with the above architecture, lmk if you prefer something else tho!